File: Difficulty: Easy/Ceil in a Sorted Array/ceil-in-a-sorted-array.cpp
// User function Template for C++
class Solution {
  public:
    int findCeil(vector<int>& arr, int x) {
        // brute
        
        int i = 0;
        int n = arr.size();
        while (i<n){
            if (arr[i] >= x){
                return i;
            }
            i++;
        }
        return -1;
            
    }
};

File: Difficulty: Easy/Ceil in a Sorted Array/ceil-in-a-sorted-array.java
class Solution {
	public int findCeil(int[] arr, int x) {
		int lo = 0;
		int hi = arr.length - 1;
		int mid;
		int ans = -1;
		while (lo <= hi) {
			mid = lo + (hi - lo)/2;
			if (arr[mid] >= x) {
				ans = mid;
				hi = mid - 1;
			}
			else {
				lo = mid + 1;
			}
		}
		return ans;
	}
}
